### ISSUE TITLE
Make sure Element Web works with Firefox ESR

### ISSUE THREAD

**Issue by olberger** [element-hq/element-web#27684]
### Your use case

It seems (given the discussion in https://github.com/element-hq/element-web/issues/27682, unless the person in charge of support there) that Firefox ESR isn't supported.

Given it has a large user base (enterprise, Debian stable and derived, etc.) it would be very valuable to have it supported.

**Comment by Tortoise17** [comment]
For me also it is broken. On firefox it is not working anymore since recent update from yesterday.

**Comment by t3chguy** [comment]
> Not sure what the best practice was at the time the supported environments were picked, but it seems not ideal today.

That would have us support Chrome down to 109, mobile browsers, Opera, KaiOS, UC Browser, and all other things we have intentionally excluded in preference to having access to sane Web APIs shortly after they are available. We have a supported environments thing at the top of our README for a reason. ESR is in a similar boat, being over a year old at this point.

**Comment by alfem** [comment]
Yes, broken in 115.12.0esr (64-bit). 

I use to report this kind of issues in mozilla channel in matrix, but unfortunately it is not possible now.

**Comment by rosemash** [comment]
I want to reiterate something that was mentioned on #27682: firefox-esr is the main supported browser in Debian stable, which is a massive user base (especially of OSS like matrix). I use the element instance at chat.mozilla.org and was completely unaware firefox-esr was unsupported until I was locked out of my only session half an hour ago. That just doesn't seem right. A debian user can install chromium, but won't be able to verify their new session with their original one.

**Comment by t3chguy** [comment]
@isAAAc Opera has never been a supported environment: https://github.com/element-hq/element-web#supported-environments

**Comment by isAAAc** [comment]
> oOpera has never been a supported environment

oh, ok ^^

> A debian user can install chromium, but won't be able to verify their new session with their original one

you could/should use the passphrase

**Comment by rosemash** [comment]
> you could/should use the passphrase

I agree, but this situation also shouldn't happen at all to necessitate that.

**Comment by oliversalzburg** [comment]
> That would have us support Chrome down to 109, mobile browsers, Opera, KaiOS, UC Browser, and all other things we have intentionally excluded in preference to having access to sane Web APIs shortly after they are available. We have a supported environments thing at the top of our README for a reason. ESR is in a similar boat, being over a year old at this point.

That's reasonable. I didn't mean to suggest to adopt `defaults`, but given that ESR is included in it, maybe it's worth considering under these circumstances. I understand not wanting to support an unreasonably large surface of clients, but including them in the Babel downleveling as a best-effort approach to increase usability, seems reasonable to me as well.

**Comment by t3chguy** [comment]
babel downleveling won't handle things like polyfilling `Intl.Segmenter` which is the cause for the recent issue

**Comment by andreymal** [comment]
> having access to sane Web APIs shortly after they are available

Why do you need them?

**Comment by t3chguy** [comment]
To avoid bloating the bundle with a mass of polyfills and points of potential supply chain attack

**Comment by andreymal** [comment]
> mass of polyfills

Why do you need polyfills? According to my grep, `Intl.Segmenter` you mentioned is not even used

**Comment by t3chguy** [comment]
@andreymal https://github.com/element-hq/compound-web/blob/15eefea5f821d546e6ea687c6da4b10af53b6f81/src/utils/string.ts#L21 & https://github.com/matrix-org/matrix-react-sdk/blob/cd39d91c15a91b8a1be930ef29d95798896a1110/src/utils/strings.ts#L87

**Comment by rosemash** [comment]
@issuehaver2 

> Adding another voice here to please add support for Firefox ESR. I had no idea this support would cease until I was suddenly locked out of the element web app yesterday, and there wasn't even an error message or anything, just a blank screen. I had to come here to find out what was going on, only to discover I can likely never access any of my old messages again. I'm stressing over this.

If you direly need to recover your old session if you're using a broken element instance, you can fix it by installing the latest build of firefox (f you're on debian there are instructions to add mozilla's repository here: https://support.mozilla.org/en-US/kb/install-firefox-linux#w_install-firefox-deb-package-for-debian-based-distributions)

Then from the new browser, go to to about:profiles, and change to the firefox-esr profile to restore your old cookies. Just make sure when you uninstall firefox-esr it doesn't remove profiles (apt remove doesn't on debian 12, I tried it) so you don't lose the session.

It's not a permanent solution but you haven't lost anything.

**Comment by t3chguy** [comment]
> No, supporting the long-term release of one of the two major engines, or vendors if you prefer, is a very specific request. It does not follow that any random browser should be supported.

The suggestion I was replying to was to use the browserl.ist `defaults` set

![image](https://github.com/element-hq/element-web/assets/2403652/be885427-b759-41d7-9fb8-cf08c1fb8755)

**Comment by oliversalzburg** [comment]
Good news. Seems like the Firefox release today is also the new ESR. https://whattrainisitnow.com/

**Comment by bleve** [comment]
Also centos-stream, rhel and all rhel clones default to firefox-esr. Please add support.

**Comment by opk12** [comment]
The Debian package status can be monitored [here on tracker.d.o](https://tracker.debian.org/pkg/firefox-esr) under `news`.

Sadly, this is the only option for the privacy-minded users, as mozilla.org's binaries are non-[reproducible](https://reproducible-builds.org/) per [bug 885777](https://bugzilla.mozilla.org/show_bug.cgi?id=885777).

**Comment by mind-zz** [comment]
Another voice encouraging "support Firefox ESR". My setup was working last week, then just suddenly stopped. A whole year might be a long time in web development terms, but it's quite short in *user terms*. I'd rather not have to be a sysadmin every time Doubleclick creates a new browser "feature".

Furthermore, why is there no option to continue using the version of Element that I had been using, before support was dropped?

**Comment by t3chguy** [comment]
> Furthermore, why is there no option to continue using the version of Element that I had been using, before support was dropped?

Because the next time your browser loads that webpage the files necessary for the app are no longer available on the webserver, only the new version is. If someone wants to contribute some service worker magic to support multiple versions then great. Also keep in mind you can't roll back from 1.11.70 because the Rust Crypto database format changed and it is irreversible.

**Comment by mind-zz** [comment]
> If someone wants to contribute some service worker magic to support multiple versions then great. Also keep in mind you can't roll back from 1.11.70 because the Rust Crypto database format changed and it is irreversible.

I'm not wanting to roll back, but rather to keep using the version that was working before the nonconsensual upgrade. In a sane world this would be doable with subdirectories for app versions, but I have no idea if that works in the current world.

**Comment by t3chguy** [comment]
> In a sane world this would be doable with subdirectories for app versions, but I have no idea if that works in the current world.

Yes a system where the path always contains the version would work though would require users to update any bookmarks they have, and going to a link in their history could corrupt their database as they'd be inadvertently rolling back. This would complicate hosting as docker hosting would go out of the window unless we baked in an increasing amount of versions into the image.

**Comment by risicle** [comment]
Lesson: if element is working for you and asks whether you want to upgrade, say *no* - especially as there doesn't appear to be any rollback mechanism.

**Comment by theres-waldo** [comment]
> Good news. Seems like the Firefox release today is also the new ESR. https://whattrainisitnow.com/

Note that a new ESR spends some time being on Debian unstable before being rolled out to Debian stable.

If I'm reading the [package tracker](https://tracker.debian.org/pkg/firefox-esr/news/?page=3) correctly, ESR115 was rolled out to Debian stable on 2024-10-07, over 3 months after the initial release (whose date according to https://whattrainisitnow.com/calendar/ was 2024-07-04).

Therefore, I don't think that "just wait for the new ESR" constitutes a solution here.

**Comment by tve** [comment]
I'm another user that got stranded with yesterdays update. Please support Firefox!
It's actually not working on vivaldi either (different problem).
I'm not using chrome/chromium for this. The Matrix is dead 👎🏻

**Comment by wetneb** [comment]
I guess I'll cancel my Element One subscription then, since I can't use it anymore? Sad, it was pretty convenient.

**Comment by BrenBarn** [comment]
> A whole year might be a long time in web development terms, but it's quite short in user terms.

That is the essence of the problem.  The stated development policies about insisting on bleeding-edge Web APIs are prioritizing developers over users.  That's a good way to build a bad reputation.

**Comment by isAAAc** [comment]
new esr version today in debian : 
```
# dpkg -l |grep firefox
ii  firefox-esr                              115.13.0esr-1~deb12u1                          amd64        Mozilla Firefox web browser - Extended Support Release (ESR)
ii  firefox-esr-l10n-en-gb                   115.13.0esr-1~deb12u1                          all          English (United Kingdom) language package for Firefox ESR
```

**Comment by t3chguy** [comment]
@isAAAc that isn't the new ESR. https://whattrainisitnow.com/release/?version=esr the current ESR is `128.0`

(Reply copied from https://github.com/element-hq/element-web/issues/27693#issuecomment-2222355317)

**Comment by langleyd** [comment]
Thanks to all the participants' thoughts and suggestions on this topic.

To recap:
- Element Web And Element Desktop’s supported environments are detailed [here](https://github.com/element-hq/element-web#supported-environments).
- While our team strives to ensure the product functions correctly on the supported browsers we can’t guarantee breakages won’t happen on other browsers, as in this case.
- **Firefox ESR, in addition to Chrome and Edge extended support releases, have not historically been included in that list. We are currently assessing the possibility of adding support for these but will need some time to conclude the feasibility of this.** 
- In the meantime, ESR 128 has now shipped. For the users affected you can find instructions to install the latest version of ESR [here](https://wiki.debian.org/Firefox#Installing_Firefox). For debian users this includes [instructions](https://support.mozilla.org/en-US/kb/install-firefox-linux#w_install-firefox-deb-package-for-debian-based-distributions) to install From Mozilla apt repository. Using these to install `firefox-esr` rather than `firefox` in the last step, launching it with `firefox-esr -P` and selecting your original ESR profile will ensure your original Element session is recovered.
- In regards to making it easier for users to know they are on an unsupported browsers:
  - We will consider clearer user documentation of the support environments linked from element.io.
  - There exists in app messaging to tell users that their current browser doesn't support the capabilities required for Element to function correctly. A bug existed that caused this to display as a white screen. This bug has now been [fixed](https://github.com/element-hq/element-web/pull/27685). We will also review the messaging and UX here. If it had worked and had directed users to updating their existing browser to the latest supported version(and we had supported the latest ESR) there would have been a sensible path forward for users.

**Comment by giplt** [comment]
> @isAAAc that isn't the new ESR. https://whattrainisitnow.com/release/?version=esr the current ESR is `128.0`
> 
> (Reply copied from [#27693 (comment)](https://github.com/element-hq/element-web/issues/27693#issuecomment-2222355317))

It is the version that comes with Debian **stable**, see https://packages.debian.org/stable/firefox-esr

**Comment by isAAAc** [comment]
can be upgraded to 128.0 by mozilla repository, https://support.mozilla.org/en-US/kb/install-firefox-linux#w_install-firefox-deb-package-for-debian-based-distributions , use firefox-esr instead of firefox when `apt install`

**Comment by Tortoise17** [comment]
I have reinstalled Firefox 128 and it worked now perfectly well so far with element on CentOS 9 Stream.

**Comment by b3** [comment]
> To recap:
[...]
>     * In regards to making it easier for users to know they are on an unsupported browsers:
>       
>       * We will consider clearer user documentation of the support environments linked from element.io.
>       * There exists in app messaging to tell users that their current browser doesn't support the capabilities required for Element to function correctly. A bug existed that caused this to display as a white screen. This bug has now been [fixed](https://github.com/element-hq/element-web/pull/27685). We will also review the messaging and UX here. If it had worked and had directed users to updating their existing browser to the latest supported version(and we had supported the latest ESR) there would have been a sensible path forward for users.

All that comes **very** late and is not enough **at all** userwise.

We were numerous committing to promote privacy Hygiene and best practices, in particular trough Tor Browser use and most notably very **strongly** through Matrix and Element clients, for instance towards our students, colleagues or any non-computer scientists (i.e. *normal*) people.

Suddendly, at least for those who start to use app.element.io as a first step they lost all their data and are enable to communicate securely since some days...

...and the only solution so far implies a lot of difficult and technical manipulations, which is a no-go for most of them.

All this shitstorm because of a policy which favors developers rather than users and prioritize bleeding-edge features over stability  :-(

Guess what will happen next?

Some clues: they will not trust Element (or Matrix because most of them do not make any differences between the protocol and the tool) anymore and it will be very difficult for now to promote it.

That is the worst case of trust loss ever observed. At least we will be able to use it to teach what not to do.

That is very sad and a real pity.

**Comment by alphapapa** [comment]
Not supporting the Firefox version shipped in Debian stable is like shooting yourself in the foot.  As others have said, it worked the other day (and has for years), and then suddenly it doesn't.

You have to realize that many users don't get to choose the software versions they run on all of the machines they use.  Organizations tend to dictate that, and they move and upgrade very slowly.  And even those who do don't always have time to deal with this.  Right now I needed Element to just work like it did the last time I used it, to send an important business message...but suddenly it doesn't work.  So now I have to go and find a workaround.  Meanwhile, my email client and server work the same as they did yesterday, and so does my phone, etc.

This is a crucial moment in Matrix's history.  Problems like this do immense damage to its reputation; it's *very* hard to un-convince people, "no, no, it's okay now, you should really start using Matrix again, trust me!"  Not one of the users--not one--cares about the underlying Web APIs.  They just need this *tool* to *keep working*.

So you really need to fix this, and fast.  The clock is ticking, and the number of people who are wondering whether they can still trust Matrix is increasing.

**Comment by t3chguy** [comment]
> Meanwhile, my email client and server work the same as they did yesterday, and so does my phone, etc.

Comparing installed apps to webapps is a bit apples and oranges. The former can leverage your dependency manager to prevent updates when you reach an incompatibility point, they can also serve different builds to different distros. These are not feasible in a webapp serving only static files.

> @ara4n Are you aware of this problem?

Matthew already responded in https://github.com/element-hq/element-web/issues/27682#issuecomment-2217590924 with the follow up management response being https://github.com/element-hq/element-web/issues/27684#issuecomment-2222807660

**Comment by tve** [comment]
> Comparing installed apps to webapps is a bit apples and oranges. The former can leverage your dependency manager to prevent updates when you reach an incompatibility point, they can also serve different builds to different distros. These are not feasible in a webapp serving only static files.

This is the weirdest comment I have read in a long time! For decades now the advantage of web apps has been that they work pretty much regardless of the end device and that improvements as well as fixes can be deployed rapidly to all users. You're trying to turn that on its head: sorry, that doesn't make sense.

**Comment by alphapapa** [comment]
> Comparing installed apps to webapps is a bit apples and oranges. The former can leverage your dependency manager to prevent updates when you reach an incompatibility point, they can also pull different builds to different distros. These are not feasible in a webapp serving only static files.

As a developer, I understand how you feel; I know the modern Web is a quagmire.  But, respectfully, not a single user of Element/Matrix cares about any of that.  Element (and Matrix) is a tool, and problems like this can wreck its reputation in a moment.  As a Matrix client developer myself, I very much want Matrix (and its whole ecosystem) to succeed.  That's why I'm trying to emphasize the significance of this issue to you.

**Comment by t3chguy** [comment]
> You deployed an update disregarding compatibility with a significant portion of your user base and you apparently lack the agility to quickly revert this update while you consider your options. That's what's going on here.

This isn't an issue of agility, it is one of policy. If the decision in management is made to support ESR then we will make the necessary changes to support it. Rolling back is not possible as the same update contained a rust crypto update which featured a migration and doing such would corrupt your rust crypto data stores, causing you to need to log out and back in.

I'm personally trying to push for ESR (and Chrome + Edge Extended Stable) to be added to our support policy but all I can do is discuss it internally and then the decision makers have to evaluate the options in front of them. The table seems to be split at the moment at least amongst the ICs, I have little insight into the thoughts of management at this time.

**Comment by t3chguy** [comment]
> And that is another point to keep in mind: these ESR releases are not "a year old" when they are maintained in a distribution like Debian. They are continually updated throughout the distro release's lifetime.

Their featureset is a year old is a more clear way of putting it

**Comment by alphapapa** [comment]
> > And that is another point to keep in mind: these ESR releases are not "a year old" when they are maintained in a distribution like Debian. They are continually updated throughout the distro release's lifetime.
> 
> Their featureset is a year old is a more clear way of putting it

A year is not very long.  Not every device can even upgrade the browser anymore.  There are tablets and smartphones filling landfills because one day the user could not load the site he used a device for anymore, even though the device itself was perfectly usable.

> I don't mind taking the flak for this given I'm the one that pushed the buttons to make the release but ultimately I'm unable to bring a resolution without buy in from some levels higher than me.

It doesn't sound like you're to blame.  I don't know if anyone even is; it probably wasn't foreseen that this would happen.  But supporting, e.g. Debian's and RHEL's browsers should be mandatory for any software that wants to be taken seriously nowadays.

**Comment by alphapapa** [comment]
(FWIW, I'd suggest pinning this issue to the top of the list.)

**Comment by obfusk** [comment]
> Rolling back is not possible

Fair enough. But I find it hard to believe there is *nothing at all* that can *reasonably* be done here to help out affected users except telling them to update their browsers. What about adding a temporary polyfill to unbreak Firefox ESR?

As I wrote in https://github.com/element-hq/element-web/issues/27682#issuecomment-2220669673:

> Given that the *immediate* issue for most users here is that Firefox ESR doesn't support `Intl.Segmenter`, would it perhaps be possible to temporarily add a polyfill for that so it starts working again? Perhaps also show some kind of warning when the polyfill is needed. That doesn't mean you have committed to support Firefox ESR indefinitely, but it would unbreak things for Firefox ESR users and allow much less painful migration to a supported browser/client.

> Then you also have more time to figure out whether Firefox ESR can be supported and make sure it's properly communicated to users whether their browser is supported or not so users can switch to a different browser if they need to.

---

> I don't mind taking the flak

I think people have been understandably frustrated at the "well it's not supported so we won't do anything to fix it" messaging from Element, mostly conveyed (though obviously not decided) by you, in these issues. That said, I don't think you're to blame here *personally*. And I very much appreciate you arguing for supporting ESR.

**Comment by t3chguy** [comment]
> Fair enough. But I find it hard to believe there is nothing at all that can reasonably be done here to help out affected users except telling them to update their browsers. What about adding a temporary polyfill to unbreak Firefox ESR?

Yes, this is likely what would be done if the decision makers deem the project wants to support ESR and other incompatible browsers. There is seemingly no synchronous polyfill you can bundle in due to the main one being WASM but there'd be a way around that.

**Comment by oliversalzburg** [comment]
I updated my initial response to the topic. I believe the desire for Debian Stable/ESR support is mostly driven from reaction. Using this version of Firefox _shouldn't_ be in anyone's best interest in the first place. If you're using Debian on your Desktop, why do you want an old browser? If you're using Tor Browser and am sensible about security, why are you using a Signal client that you don't control? I know it sucks, but you have to re-evaluate if this impacts you.

**Comment by obfusk** [comment]
> If you're using Debian on your Desktop, why do you want an old browser?

NB: This is getting OT.

Few people want "an old browser". Many people want or need a *stable computing experience*, not having to deal with frequent changes in software they depend on. That's the whole point of Debian stable. And it's very much a *supported browser*.

Now, I do think it would be nice to have regular Firefox available in Debian stable as well, for people that prefer the latest and don't mind frequent changes. Because I'd rather get that from Debian, where everything is build from source and ideally also reproducible. But for a lot of people, especially less technical users like my elderly relatives, stability is important. Changes can affect your workflow or getting used to. And Firefox ESR is a perfectly reasonable choice.

**Comment by giplt** [comment]
Completely agree. Sure, there are still ways to get to the matrix. But this is rather about the less tech savvy people who don't need bleeding edge features but DO have ethical considerations around using software.

I have been recommending Debian stable as the secure and trustworthy one-stop-go for the most common use cases of communication, office work and web browsing. I also recommended Matrix/Element as the best alternative to proprietary chat applications. And with some success as a fair number of communities I am involved in now use Element for their everyday communication. All of a sudden this combination doesn't work anymore.

Hard as it is to help people navigate software ethics, Element effectively deprecating major open source platforms doesn't help. Most people will never find this github issue and/or tinker their apt repo source lists. I urged people not to judge harshly on the occasional encryption glitches they experience. Just update regularly and things will (and did) improve. But silently withdrawing support and offering a blank web page will make some simply give in to peer pressure and switch (back) to big tech.

**Comment by BrenBarn** [comment]
> This isn't an issue of agility, it is one of policy. If the decision in management is made to support ESR then we will make the necessary changes to support it. Rolling back is not possible as the same update contained a rust crypto update which featured a migration and doing such would corrupt your rust crypto data stores, causing you to need to log out and back in.

That is indeed a problem of policy, but that is still a problem and this issue has drawn attention to it.  The general impression I get here is one of backwards decision-making.  Why is the policy not to support ESR?  Who is making that policy?  What is the goal of such policies?  It seems to me the way these policies should be made is sort of like this:

1. What audience do we want to reach?
2. What are the characteristics of that audience?
3. What tools do they use?
4. How can we help them, or at least not hurt them?
5. What testing practices are needed to ensure changes don't hurt users?

Instead it seems that platforms were picked for support just because that's what's convenient to develop and/or test on, and then that resulted in not testing on platforms *that actual users actually use*.  But that indicates that the policy-making process is not grounded in something like "what are the things users do, let's be sure we support that".
 
As was mentioned in other comments, every incident like this does irreparable harm to the reputation not only of Element but of Matrix as a whole.  It doesn't really matter what your policies are about what platforms are supported.  If users are hurt, they will turn away from your product, more so the more painful the hurt is.  Losing access to all of your messages is pretty painful, so something like this has the potential to alienate many users.  (Again, not just to alienate them from Element, but to alienate them from Matrix.)

Support policies should be driven by user needs, not the other way around.

**Comment by obfusk** [comment]
To make a slightly unfair comparison: Android 13 is very much "over a year old" too. And yet I would be very surprised to find the Element Android apps don't work on anything but very recent Android versions. And indeed Element X supports Android 7.0+ and Element supports Android 5.0+. Those are quite a bit older than Firefox ESR.

**Comment by wodny** [comment]
@ oliversalzburg:
> In fact, if you _only_ install packages from Debian stable repositories, you're leaving yourself vulnerable to issues that stand unfixed in the stable versions of Debian. _Stable_ in Debian terms does not mean more secure or _better_ in any way, especially if you're on Desktop.

This is [false][1]. The stable Debian is always the one that gets timely security updates provided by the [Security Team][2]. On the other [hand][3]:
> [Debian Unstable](https://wiki.debian.org/DebianUnstable) - repository where new & untested packages are introduced.
> [Debian Testing](https://wiki.debian.org/DebianTesting) - repository with packages from unstable, if no bug are found within 10 days.

For example compare how OpenSSH [CVE-2024-6387][4]  and XZ [CVE-2024-3094][5] were handled and which releases were affected.

[1]: https://wiki.debian.org/DebianSoftware
[2]: https://wiki.debian.org/Teams/Security
[3]: https://wiki.debian.org/DebianStable
[4]: https://security-tracker.debian.org/tracker/CVE-2024-6387
[5]: https://security-tracker.debian.org/tracker/CVE-2024-3094

**Comment by rosemash** [comment]
@oliversalzburg @wodny 
I fear this may be a common misconception and I just want to spell it out even clearer.
ESR stands for Extended Support Release **because** it receives timely updates and security fixes and fixes critical bugs. firefox-esr isn't outdated, it is a mainline branch that is officially supported by both Debian and Mozilla. It gets updates and fixes just as frequently as the main branch does, sometimes even faster.
You'd think people in an OSS setting would understand stable release cycles, but if you go back through the thread, it's clear there are a lot of people who just assume the best practice for Debian is to meddle with apt sources and update some packages to newer versions when appropriate.

**Comment by obfusk** [comment]
> ESR stands for Extended Support Release **because** it receives timely updates and security fixes and fixes critical bugs. firefox-esr isn't outdated, it is a mainline branch that is officially supported by both Debian and Mozilla. It gets updates and fixes just as frequently as the main branch does, sometimes even faster.

Exactly. If you look at https://www.mozilla.org/en-US/security/advisories/ you can see both regular and ESR Firefox getting fixes at the same time. It just doesn't get new features as often. But that doesn't make it "old".

**Comment by jonorthwash** [comment]
Wow.  I just want to add this is one of the weirdest FOSS issues I've seen in several decades of using FOSS.  The devs breaking support for ESR software, adamantly refusing to find a work-around, and not even writing a little handling to that users receive a message about it being an incompatible version?

The last part is the weirdest part of this.  The in-browser Element client has detection for lack of javascript support in the browser, but doesn't tell you "sorry we don't support Firefox ESR".  Instead one day your client says there's a new version available, you let it reload, and you just get a blank page that no amount of clearing cache or using private browser windows will make change.

I expect an occasional glitch or something when using Firefox ESR, like I get in corporate products (looking Google's direction—and they occasionally even have messages saying that they don't really support Firefox), and I expect FOSS devs to be overworked and not always support a particular configuration, but a blank page with no further explanation for all Debian stable and Tor users?  It's the worst of both worlds.

**Comment by jonorthwash** [comment]
> Wow. I just want to add that this is one of the weirdest FOSS issues I've seen in several decades of using and developing FOSS. The devs breaking support for ESR software, adamantly refusing to find a work-around, and not even writing a little error handling so that users receive a message about it being an incompatible version?

I believe some of my surprise here is exactly because of what @rosemash stated.  Element devs clearly have no idea what ESR means, and treat it simply as "old".  There's a disconnect between the development team and the real world that is causing them to fail to understand why this issue is such a big deal.  Their attitude has been dismissive from the start.

Devs: take pride in your work, and don't dismiss your users when they tell you that you screwed up.

**Comment by t3chguy** [comment]
> and not even writing a little error handling so that users receive a message about it being an incompatible version?

That has been done, it was buggy in the release but the fix will ship in the upcoming release.

> Element devs clearly have no idea what ESR means, and treat it simply as "old".

Element devs follow the policy set out by people that manage them, they get an input but they do not control the policy.

**Comment by jonorthwash** [comment]
> > and not even writing a little error handling so that users receive a message about it being an incompatible version?
> 
> That has been done, it was buggy in the latest release but the fix will ship in the upcoming release.

Why isn't this a patch given more priority?  How many duplicate bug reports have you received already because this is still just failing silently?  And those are just the users who know enough, have a GitHub account, and care.  The rest are just getting confused and assuming the problem is either their "computer" or your product.

**Comment by jonorthwash** [comment]
> > Element devs clearly have no idea what ESR means, and treat it simply as "old".
> 
> Element devs follow the policy set out by people that manage them, they get an input but they do not control the policy.

How do we get a message to these people that manage the devs?  Surely they need a policy that's in line with ESR browsers and that avoids alienating users by giving them a blank page.

**Comment by t3chguy** [comment]
> Why isn't this a patch given more priority?

I fixed it the same day of the first report

**Comment by jonorthwash** [comment]
> > Why isn't this a patch given more priority?
> 
> I made the patch the same day of the first report

A patch given more priority would've been rolled out then too.  So while this information is helpful, it doesn't address the question you appear to be responding to, which I can now make more specific: why has the patch still not been pushed yet?

**Comment by t3chguy** [comment]
> why has the patch still not been deployed yet?

It has been deployed to develop.element.io and staging.element.io. It will be deployed to app.element.io when the release happens later today. 1.11.70 ran late due to issues with rust crypto and this meant there was no gap between 1.11.70 and the following RC for any hotfix releases, our tooling currently only supports maintaining one active release cycle/branch at a time and as the policy stands the issue only affecting (currently) unsupported environments it wasn't a drop everything and make the release process happen sort of thing, as that'd have knock on effects on customer projects and everything else.

**Comment by gpcf** [comment]
> Wow. I just want to add that this is one of the weirdest FOSS issues I've seen in several decades of using and developing FOSS. The devs breaking support for ESR software, adamantly refusing to find a work-around, and not even writing a little error handling so that users receive a message about it being an incompatible version?

Honestly, I'm completely baffled at this attitude, as Element has marketed itself quite a bit as a messenger for institutions. I work in IT at a large German university, and basically the entire department runs on Debian, while the University has set up element as a chat service, and I think I am very much not alone with this kind of setup. This is an issue that affects thousands of users, many of them too non-technical to use the "fixes" suggested by the devs, such as students using a university computer pool. The way this was handled is how you lose users to other platforms, and lose them for good.

**Comment by galitus** [comment]
I second the post above and would like to add, myself an admin for a bunch of PCs/Laptops, I use the ESR for deployment.
Why? Because I don't want to deploy every second week a new release.
It is 30 releases from 115.1/116  to 128 for the mainline version and only 16 for the ESR one. 
And what features do normal people need from the mainline version anyway, to exclude the ESR here?

I like the product matrix/element, and it hurts to see that this kind of development hurts the acceptance and usage in the long run.

**Comment by jonorthwash** [comment]
I now get the "Unsupported browser" message.  Which just tells me to "Please install [Chrome](https://www.google.com/chrome), [Firefox](https://firefox.com/), or [Safari](https://apple.com/safari) for the best experience."  While running in Firefox.  Stating "latest version" might help avoid even more confusion.

**Comment by t3chguy** [comment]
@jonorthwash it is on Design's todo list to redesign that view with better messaging, my fix merely made the existing one work again.

**Comment by aabittner** [comment]
my 2ct: wonder about this whole situation and people in charge of this project seem? or act as if there didnt know about firefox and their channels and how the whole software industry works out there in the real world, at least thats my taking on this situation. firefox esr channel has been there for years, and obviously the real world doesnt update a piece of software the moment something is released to the public, and there are overlapping time spans of software versions in the real world as well. when this whole matter started of suddenly deprecating the then-current firefox esr 115.x it was absolutely cler that there was no released esr beyond that version range, and esr 128 only appeared a day or two later into this bug, then people in charge in here pointing out merely that firefox (esr) did in fact appear as 128 that very day. anyhow, in reality web developers should know about big browser vendors and about projects making use of these browsers and how they ship their stuff and that they dont abandon an esr channel for example that is absolutely designed and laid out in the way to be overlapping at the beginning/end of the releases obviously.

firefox 115 esr will coexist for several more months now in probably quite a number of linux distros and other situations and surroundings out there (academia, huge institutions etc were some exmples here in this bug ticket) and this is still not a good solution on my opinion when people just cant switch over to esr 128 for a long time just yet.

please consider this matter and this bug for the next installment of ESR of firefox in the future. it is all laid out and known in advance as of today. if plans dont change fundamentally, this will be the same situation again at the next ESR switching to a higher major release level.

thanks.

**Comment by konomikitten** [comment]
Just an FYI but clicking the `Go to element.io` link does not work on the new "Unsupported browser" page.

User Agent: `Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/115.0`

**Comment by MatMaul** [comment]
Any chance the changes get reverted for a couple of months only ?

The new ESR is out, but the old one will still be around for some time, until admins upgrade it.

We are heavily impacted on Tchap federation too, and we will have to revert this changes on our side, quite a lot of state deployments use ESR.

**Comment by Seirdy** [comment]
[The Tor Browser has historically lagged on ESR updates](https://seirdy.one/notes/2022/09/19/state-of-the-tor-uplift/) due to the amount of patching required, so if anonymous users are an important demographic it's generally a good idea to support an ESR version for a few months after it's EoL. It sucks to have to support EoL software but unfortunately there are no other browsers that provide comparable fingerprinting protections.

**Comment by theres-waldo** [comment]
> Any chance the changes get reverted for a couple of months only ?

Just to make this request more explicit: I know that it was explained in a [previous comment](https://github.com/element-hq/element-web/issues/27684#issuecomment-2225893028) that a wholesale revert to a previous Element version is not feasible.

But has reverting just the specific change that introduced the ESR incompatibility (which, if I understood the disucssion correctly, was the introduction of a dependency on `Intl.Segmenter` in https://github.com/matrix-org/matrix-react-sdk/pull/12617) been considered?

It seems to me that such an action could be taken, to address the current pains in the short term, independently of any larger discussion / commitment about ESR support going forward.

**Comment by t3chguy** [comment]
> But has reverting just the specific change that introduced the ESR incompatibility (which, if I understood the disucssion correctly, was the introduction of a dependency on Intl.Segmenter in https://github.com/matrix-org/matrix-react-sdk/pull/12617) been considered?

That would not fix your issue, as some dependencies, e.g. https://github.com/element-hq/compound-web also use Intl.Segmenter so the same exception would be thrown. The teams responsible for those dependencies would also need to make similar changes.

> It seems to me that such an action could be taken, to address the current pains in the short term, independently of any larger discussion / commitment about ESR support going forward.

Yes an action could be taken, if we were tasked with such, as the end of the day our employer chooses what we work on.

**Comment by erebion** [comment]
> In fact, if you _only_ install packages from Debian stable repositories, you're leaving yourself vulnerable to issues that stand unfixed in the stable versions of Debian.

The Debian maintainers do a lot of hard work to fix security issues. Please do not spread FUD.

Aside from that, ESR is not only available for Debian.

Many people have given a lot of excellent examples in response, so I will not add any more.


> But has reverting just the specific change that introduced the ESR incompatibility (which, if I understood the disucssion correctly, was the introduction of a dependency on `Intl.Segmenter` in [matrix-org/matrix-react-sdk#12617](https://github.com/matrix-org/matrix-react-sdk/pull/12617)) been considered?
> 
> It seems to me that such an action could be taken, to address the current pains in the short term, independently of any larger discussion / commitment about ESR support going forward.

Yup. It could then be re-introduced ones ESR has moved to a newer version with that included.

> That would not fix your issue, as some dependencies, e.g. https://github.com/element-hq/compound-web also use Intl.Segmenter so the same exception would be thrown. The teams responsible for those dependencies would also need to make similar changes.

Seems sensible. Do not lock out a large amount of users and for that just delay that a bit until ESR has caught up.

**Comment by o5k** [comment]
> [...] if you only install packages from Debian stable repositories, you're leaving yourself vulnerable [...]
> [...] your desire to use an ancient ESR [...]

ESR is updated. It is by objective definition not ancient. That's its whole thing. Don't play the already jaded "old = bad" card when the software in question isn't even old. It's getting _very_ tiring to hear this rhetoric of "if you're not running the latest nightly builds you are a luddite".

**Comment by o5k** [comment]
> my 2ct: wonder about this whole situation and people in charge of this project seem? or act as if there didnt know about firefox

@aabittner yeah, that tends to be how it works — a dev pushes an update that breaks compatibility first, _then_ they yap and complain about how everyone having issues is a luddite who needs to get with the times and use the Latest and Greatest
you _never_ get week-long defensive back and forths from developers who actively knew what they were doing when pushing a breaking change

**Comment by srett** [comment]
\*sigh* Matrix being Matrix again. I wish we'd use IRC at work, but I guess without emoji reactions you can't be productive in this day and age...

**Comment by timmc** [comment]
Besides the concerns others have raised here around minimum support and communication, I would implore the devs to take more care to highlight breaking changes in the changelog. I review the changelog every time I go to upgrade my shared Element installation and I rely on it for such warnings.

**Comment by erebion** [comment]
What was the last compatible version? I struggle finding out where exactly it broke.

**Comment by timmc** [comment]
1.11.69 is the last one that worked for me. I upgraded our installation 1.11.71 and was confronted with a weird error message from the browser about security headers or something, then tried 1.11.70 and got a different error. Downgrading to 1.11.69 worked, presumably because I didn't get far enough in to have my session corrupted.

**Comment by BrenBarn** [comment]
> Besides the concerns others have raised here around minimum support and communication, I would implore the devs to take more care to highlight breaking changes in the changelog.

I'm not a dev, but my interpretation of their comments here is that this isn't considered a breaking change due to the platform support policy.  There is no commitment to keeping the app working on any particular browser version that it used to work on.  It's just "the last two versions of major browsers".  If new browser versions come out, the old ones aren't supported anymore, so there's not even any attempt to test on those versions, and changes that break them aren't considered breaking.

Personally I deplore this kind of development philosophy and (like I mentioned in an earlier comment) I think it speaks to misguided decision-making at a higher level (see also repeated comments from the devs saying "this was a management decision").

**Comment by timmc** [comment]
I agree that that might be the philosophy, but I've seen any number of projects note breaking changes for "unsupported" cases.

**Comment by knarrff** [comment]
Independent of the current mess: can we please get ESRs into the list of supported environments for Element?

Also, just because there is a new ESR out does not mean the old is not supported anymore. There has to be a grace period for distributions and users alike, e.g. while 128 was released 2024-07-09, the EOL of the last EST before that (115) is 2024-10-01. 115 should be supported at the very least until that date by element.

**Comment by srett** [comment]
Fixed with userscript

```js
// ==UserScript==
// @name Fix the Matrix
// @match https://<your homeserver>/*
// @run-at document-start
// ==/UserScript==

Intl.Segmenter ??= class { *segment(string) { for (const segment of string) yield {segment}; } };
```

Works in ViolentMonkey but not GreaseMonkey because webdev.

**Comment by obfusk** [comment]
```js
Intl.Segmenter ??= class { *segment(string) { for (const segment of string) yield {segment}; } };
```

Whilst it's a clever trick to unbreak Element not even loading, this isn't an actual replacement for what `Intl.Segmenter` does. It will produce incorrect results for anything but basic English. It will not handle emoji or any kind of character represented using multiple code points.

It might be a viable workaround for some people but it is not a "fix". I wish it was that easy. I thought of doing something similar but even if I tried to take variation selectors and zero width joiners into account there would be far too many cases where it produces the wrong result, even if it would likely work for me.

Which is of course exactly why Element uses `Intl.Segmenter`. It's absolutely the right choice. Or would be if it was supported in Firefox ESR.

**Comment by andreymal** [comment]
>a viable workaround for some people

This is called “graceful degradation”, a now-forgotten technique of ancient web developers.

**Comment by obfusk** [comment]
> This is called “graceful degradation”, a now-forgotten technique of ancient web developers.

Hopefully, yes. But I don't know what the consequences of the workaround producing wrong results are here. If it's just going to result in rendering some text badly or could actually cause nasty bugs.

**Comment by srett** [comment]
> Whilst it's a clever trick to unbreak Element not even loading, this isn't an actual replacement for what `Intl.Segmenter` does. It will produce incorrect results for anything but basic English. It will not handle emoji or any kind of character represented using multiple code points.

No, this is a complete reimplementation that I thoroughly validated with millions of unit tests.

> It might be a viable _workaround_ for some people -- if you really don't need correct results for all those other cases like emoji -- but it is not a "fix". I wish it was that easy. I thought of doing something similar but even if I tried to take variation selectors and zero width joiners into account there would be far too many cases where it produces the wrong result, even if it would likely work for me.
> 
> Which is of course exactly why Element uses `Intl.Segmenter`. It's absolutely the right choice. Or would be if it was supported in Firefox ESR.

It did just fine without `Intl.Segmenter` until two weeks ago. The only place where I noticed it being used is those fallback avatars that contain the initial letter of the user. So yes if you don't have an avatar set and your nick starts with an elephant, this might not display correctly. I guess between potentially displaying garbled avatars and not supporting Firefox ESR, the latter is probably the right choice.

But most likely nobody is even really aware what else this might be used for, since according to [that comment above](https://github.com/element-hq/element-web/issues/27684#issuecomment-2217256568), it's not to be found in element's source code directly but, as usual with modern web apps, in a dependency of a dependency of a dependency that's a dependency of left-pad.

I can't rule out this somehow manages to kill your dog or something. But at least when you use that hack, **Element still displays a warning that your browser might not properly be supported and you need to acknowledge this by clicking a button.** I guess that's something and would make it acceptable to officially include this (at least as a hotfix release, in case someone want's to properly tackle this). As said, *the disclaimer is already there*. But picking a pragmatic solution isn't cool these days, I know. "Users won't understand that warning, dismiss it and then complain about garbled avatars!"

**Comment by giplt** [comment]
Could you perhaps add a link to https://matrix.org/ecosystem/clients/ in the error message that is displayed? Until Element gets fixed users can navigate to an alternative web client that will work in their browser without having to upgrade their systems...

**Comment by obfusk** [comment]
> It did just fine without Intl.Segmenter until two weeks ago.

Yes. Because it was using a library to do the same thing. One that produces correct results because it actually splits on graphemes, not code points like your code.

> I guess between potentially displaying garbled avatars and not supporting Firefox ESR, the latter is probably the right choice.

I agree with that 100%.

> But most likely nobody is even really aware what else this might be used for.

And that was my point. That we don't know for certain what might break with a replacement that doesn't actually do what `Intl.Segmenter` does. I'm not saying "don't use it". I'm just saying *tell people that the most likely breakage will be something harmless like garbled avatars but that no one knows for sure*.

**Comment by rosemash** [comment]
> Yes. Because it was using a library to do the same thing. One that produces correct results because it actually splits on graphemes, not code points like your code.

I don't see any reason element couldn't just continue to use the library it was using before. It seems like there was no reason to hastily update to a bleeding edge browser implementation of something that was previous done by a library. Furthermore, there's no reason it can't return to using the library to continue to support Debian.

**Comment by intrigus-lgtm** [comment]
I can also offer [this script](https://gist.github.com/intrigus-lgtm/43f0b46680ff66cdc9ad2dabf2643547) which uses https://github.com/cometkim/unicode-segmenter?tab=readme-ov-file#export-unicode-segmenterintl-polyfill to polyfill Intl.Segmenter.

**Comment by andreymal** [comment]
> ... avoid bloating the bundle with a mass of polyfills and points of potential **supply chain attack**

Great, now users put random, potentially insecure scripts from the internet instead

**Comment by knarrff** [comment]
> Great, now users put random, potentially insecure scripts from the internet instead

It is a one-line script and not difficult to check and understand. And that line fixes (or at least works around) an issue the a vendor isn't willing to fix. I think we wouldn't have half of this discussion if by now there would be the understanding that the element devs would recognize that there are a lot of users out there that aren't living on bleeding edge and don't want to and have their own good reasons, and they are users that deserve support, too.

**Comment by t3chguy** [comment]
The devs understand this. We're still awaiting management to make the final decision here.

**Comment by andrewshadura** [comment]
What prevents you from reverting the PR which introduced this regression now, and then wait for the final decision?

**Comment by t3chguy** [comment]
> What prevents you from reverting the PR which introduced this regression now, and then wait for the final decision?
> 

Reverting two prs and doing releases on four projects will eat up most of a day. Can't do that without management signing off on that time.

**Comment by andrewshadura** [comment]
So a week of time of your users is eaten instead. Thanks!

**Comment by t3chguy** [comment]
> So a week of time of your users is eaten instead. Thanks!

Sorry I'm not going to insubordinate because a user asks me to be.

**Comment by erebion** [comment]
> Reverting two prs and doing releases on four projects will eat up most of a day. Can't do that without management signing off on that time.

What would be the best way to let management know about the issue and kindly ask them to allow it to get fixed?

**Comment by t3chguy** [comment]
They are aware hence the response here much earlier. It is going through the processes, you just have to wait for them to come back to you with a decision.

**Comment by alphapapa** [comment]
> They are aware hence the response here much earlier. It is going through the processes, you just have to wait for them to come back to you with a decision.

Respectfully--and not to criticize you or point fingers at anyone in particular--this seems like an "all hands on deck" kind of decision that ought to be made within a day or so.  For every person complaining here about breakage, there are who-knows-how-many others who just stopped using Matrix instead, because they neither know nor care about this place and how Matrix works behind the scenes.  The potential damage to Matrix's long-term reputation caused by this breakage--I think it can hardly be exaggerated.  The urgency here ought to be something like one level below "emergency."

**Comment by obfusk** [comment]
AFAIK ["the Element devs are stuck doing other paying-customer stuff as their day job"](https://github.com/element-hq/element-meta/issues/339#issuecomment-2057475632). So without a paying customer, I doubt this will get fixed any time soon. I personally think it's absurd that this isn't considered high priority, but devs don't owe anyone free labour and Element as a for-profit company has it's own priorities that do not necessarily align with ours. I do wish they'd be more upfront about that, a "sorry we'd like to fix this but we can't pull people off their day jobs to do it" would still suck but at least it wouldn't just leave users waiting.

**Comment by jgoerzen** [comment]
I would just like to add here that Firefox ESR is not just for Debian.  It's really a key part of [Mozilla's enterprise strategy](https://www.mozilla.org/en-US/firefox/enterprise/).   Firefox has two update channels: "rapid release" and ESR.  Firefox 115 IS one of the two most recent ESR releases (though not one of the two most recent Rapid releases).  Note that ESR and Rapid releases are not actually the same, even on day zero when they come out.

Why is ESR important?  Imagine you're supporting 100,000 clients.  Your users run various mission-critical webapps.  They may have custom plugins.  If you can only get security updates by upgrading, that is a nightmare; some of the apps may be broken by updates, etc.  So you have ESR.  These organizations can still maintain security while having a more achievable 42-week window for validating and updating.

Another scenario would be: consider if you're supporting not very computer-literate relatives at a distance.  Grandparents, say.  Are you going to give them something that might change how it looks and behaves every month, or something that you can upgrade for them when you see them twice a year (and stay secure between that anyhow)?

My point here is that as Element is trying to sell the Matrix ecosystem into businesses -- something I wish them much success with -- they are not going to be able to do so with the kind of attitude evinced here.  Enterprises simply cannot easily upgrade their browser to new feature releases every month.  Not only that, but 115 ESR *IS* one of the two most recent Firefox releases.

**Comment by l0b0** [comment]
One big reason to run ESR is [policy templates](https://mozilla.github.io/policy-templates/), many of which are not (and are not expected to ever be) supported on mainline Firefox; see for example [this discussion](https://github.com/mozilla/policy-templates/issues/706). I'm running ESR to have this much more sane way of configuring Firefox than profile.json, and I expect so are many workplaces where enterprise profiles are in use.

**Comment by t3chguy** [comment]
> So you have ESR. These organizations can still maintain security while having a more achievable 42-week window for validating and updating.

I do wish that ESR would stick to its 42 week window though. 

[image](https://github.com/user-attachments/assets/0d79ebb2-a078-42bf-9e8f-0a08999fcfef)

The above is between ESR115 and ESR128

It would certainly unblock a lot of things. If we ended up needing a feature which ESR lacked and had to tell a customer, sorry we can't ship it for until next year or maybe the year after that would likely sink a deal. Not having actual dates to work off like other LTS offerings (Chrome & Edge Extended Stable) is a bit crud. And this is also ignoring the (as far as I can tell) lack of a solid timeframe of how long distros like Debian need to ship a given ESR from the moment it lands. If we could say its 42+12 weeks and never a moment more that'd at least be a starting point.

**Comment by theres-waldo** [comment]
> Not having actual dates to work off like other LTS offerings (Chrome & Edge Extended Stable) is a bit crud.

Firefox ESR has been on a pretty stable annual cadence, with EOL dates being around September of a given year:

 * Firefox 91 EOL: 2022-09-19
 * Firefox 102 EOL: 2023-09-25
 * Firefox 115 EOL: 2024-09-30

(I'm basing these on https://whattrainisitnow.com/calendar/, with the EOL date being the last day of the release cycle of the last minor version for a given ESR.)

**Comment by dadinn** [comment]
![image](https://github.com/user-attachments/assets/428240ee-df68-4a37-8f87-3eca4119414e)

Wait, are you saying this is really just a vendor lock-in product disguised as OSS?
The FLOSS community is essentially their guinea pig beta-testers.

**Comment by t3chguy** [comment]
> Firefox ESR has been on a pretty stable annual cadence, with EOL dates being around September of a given year:

Right, so their "average 42 weeks" is a farce? https://support.mozilla.org/en-US/kb/choosing-firefox-update-channel

> receives major updates on average every 42 weeks with minor updates such as crash fixes, security fixes and policy updates as needed, but at least every four weeks.

**Comment by andrewshadura** [comment]
@t3chguy, I just wanted to point out comments like the above you made are very inappropriate and make a really bad impression about Element (an impression I hope isn’t true).

@ara4n, how come this has been dragged along for two weeks? This is extremely disappointing.

**Comment by knarrff** [comment]
> Right, so their "average 42 weeks" is a farce? https://support.mozilla.org/en-US/kb/choosing-firefox-update-channel

A comment about ESR releases of someone else being later than originally planned, by someone who doesn't even have them? Really?

**Comment by t3chguy** [comment]
> @t3chguy, I just wanted to point out comments like the above you made are very inappropriate and make a really bad impression about Element (an impression I hope isn’t true).

I think I'm granted my own opinions outside of my working hours

**Comment by andrewshadura** [comment]
Anyone can have any opinions, but that does not mean the comments they make may be inappropriate and rude, which that comment undoubtedly was.

**Comment by erebion** [comment]
Just calm down, everyone. Generating more comments than people can read will not help us get the issue solved.

**Comment by AdamMajer** [comment]
Enterprise distros would generally follow ESR. ESR has overlap of support upstream allowing for migration from one ESR to next. This happens for any software with long term support. For firefox, there are currently, 

Firefox 115.13.0esr **and**
Firefox 128.0esr

**Comment by theres-waldo** [comment]
> > Firefox ESR has been on a pretty stable annual cadence, with EOL dates being around September of a given year:
> 
> Right, so their "average 42 weeks" is a farce? https://support.mozilla.org/en-US/kb/choosing-firefox-update-channel
> 
> > receives major updates on average every 42 weeks with minor updates such as crash fixes, security fixes and policy updates as needed, but at least every four weeks.

That does seem to be out of date. I filed https://bugzilla.mozilla.org/show_bug.cgi?id=1909292 to get it updated.

**Comment by ptman** [comment]
https://riots.im/ can be handy for using older element-web versions if you don't want to serve an older version yourself (tarball + https server)

**Comment by langleyd** [comment]
Thanks again to all for the feedback and thoughts on this topic.

Until now Firefox ESR was outside of our support policy, and was therefore not a browser we tested for. This coupled with a recent breakage of the “Unsupported Browsers” screen, meant users were left in a broken state. Apologies to those users left in this state during the last release while a decision was made on the possibility of adding support.

Element has decided to add support on a _best effort_* basis for the latest Firefox ESR and Chrome/Edge Extended Stable browsers, the timeframe for which will be extended until the new version of Firefox ESR lands in Debian stable plus one additional release cycle(2 weeks). We have fixed the Unsupported Browsers screen and also and have also added a [code change](https://github.com/element-hq/element-web/pull/27803) for the upcoming Release Candidate that will allow users on Firefox ESR 115 to continue to the product from the Unsupported Browsers screen. We will also be improving the in-app experience to better inform users ahead of time that support for their browser will cease.

\* What does “best-effort” mean in practice?
- The wider Element Products(including Element Call and the Enterprise Server Suite) do still not officially support these browsers. Element Call currently works with Firefox ESR 128 but a break in the future may be unavoidable (E.g. If Element called needed to use a modern api to support End-to-End encrypted VoIP).
- The element web project and its contributors should keep the client functioning and gracefully degrade where other sibling features like Element Call may not function.
- We will invest in tooling and processes to try ensure the extended support browsers continue to function for the period outlined above though this may be challenging; for example our primary tool for such testing, Playwright, does not currently support older Firefox releases (see[ https://github.com/microsoft/playwright/issues/31163](https://github.com/microsoft/playwright/issues/31163)).

**Comment by alphapapa** [comment]
@langleyd Thank you!

**Comment by wodny** [comment]
> > > Firefox ESR has been on a pretty stable annual cadence, with EOL dates being around September of a given year:
> > 
> > Right, so their "average 42 weeks" is a farce? https://support.mozilla.org/en-US/kb/choosing-firefox-update-channel
> > > receives major updates on average every 42 weeks with minor updates such as crash fixes, security fixes and policy updates as needed, but at least every four weeks.
> 
> That does seem to be out of date. I filed https://bugzilla.mozilla.org/show_bug.cgi?id=1909292 to get it updated.

The SUMO page has been updated to declare 52 weeks on average.

**Comment by TurtleWilly** [comment]
> * 11 of these instances are currently not updated automatically/periodically anymore

I had to stop updating after 1.11.26, because compatibility with Firefox 78.15.0esr was broken. (That's the most up-to-date browser I can run w/o having to buy new (Mac) hardware). 😎

Personally I would expect better in regards of backwards compatibility from both Element (should support older ESR releases too) and Firefox (should support older versions of (mac)OS too).

All this insane level of bleeding edge and security theater forces poor folks like me to run with way older version that are like "not secure" at all. What is this madness? What is the most strange thing to me: these days this is mainly driven by OSS based projects. There's a lack of community approach regarding backwards compatibility that avoids discriminating people that just can't afford this rat's race. 😱 Oh and dare when you report an issue in such case then you'll most likely be looked at like you're crazy and come from Mars. I'm quite sure some will raise an eyebrow about my comment here too: "WTF doe this person wants with Firefox 78ESR support now?". 

So what? You have to provide a few extra bytes of polyfill to support the **latest** Firefox ESR (115)? What's the big deal? Why is that something even to not consider? I do not understand this at all. If I was in charge of Element it certainly still would work perfectly fine with Firefox ESR 78 (probably even a step older).

Anyway… just to give you another perspective… 😇

**Comment by t3chguy** [comment]
> So what? You have to provide a few extra bytes of polyfill

Not everything is polyfillable, e.g. ESR 115 won't support anything relying on WebRTC insertable streams as a polyfill is not possible. 

![image](https://github.com/user-attachments/assets/a70567a3-9a66-4f79-b3d8-dd081564f2e3)

Which renders things like end-to-end encrypted video conferencing broken

**Comment by olberger** [comment]
Wow, this was by far the longest thread under an issue I ever filed ;-)

Thanks for the responses from the management, and hopefully, most of the participants learned something in those eventful times.

**Comment by ghost** [comment]
I just set up a new machine with Debian 13 and got this screen:

<img width="386" height="199" alt="Image" src="https://github.com/user-attachments/assets/88cf2e82-9004-4559-9af0-502f054bb488" />

Firefox 128.13.0esr

It seems to work though it tells me "unsupported". I just wanted to let you know :).

Have a nice day.

**Comment by t3chguy** [comment]
Yup. It isn't supported. It'll work best-effort as per https://github.com/element-hq/element-web#supported-environments

**Comment by knarrff** [comment]
> Yup. It isn't supported. It'll work best-effort as per https://github.com/element-hq/element-web#supported-environments

Yes, and this is what this ticket is about. "The last two major versions of Firefox" should mean "the last two ESR versions", or at least "the last ESR version". And 128.13 **is** the current ESR version, and with that **should** be supported. There are good reasons to provide and use an ESR version, but that all goes down the drain when developers of websites always insist on the newest and most recent.

**Comment by t3chguy** [comment]
@knarrff you can read the response from management earlier in the thread. Not every feature request gets implemented as demanded by the community. You can always make contributions, maintain a fork or sponsor the support of ESR.
